librecad/src/lib/math/rs_math.cpp

```diff
@@ -239,11 +240,99 @@ bool RS_Math::isSameDirection(double dir1, double dir2, double tol) {
 	return getAngleDifferenceU(dir1, dir2) < tol;
 }
 
+
+/*
+    Parse for units in expression, and then convert their values into millimeter.
+
+    - by (2021) Melwyn Francis Carlo <carlo.melwyn@outlook.com>
+*/
+QString normalizedUnitsExpression(QString inputExpression)
+{
+    QRegExp regexNotAlpha("[^a-zA-Z\"\']*");
+
+    if (regexNotAlpha.exactMatch(inputExpression)) return inputExpression.trimmed();
+
+    inputExpression = inputExpression.trimmed() + " ";
+
+    bool previousCharacterWasLetter  = false;
+    bool previousExpressionWasNumber = false;
+
+    QString outputExpression  = "";
+    QString unitsBufferString = "";
+    QString valueBufferString = "";
+
+    for (int i = 0; i < inputExpression.length(); i++)
+    {
+        const QChar testCharacter = inputExpression.at(i);
+
+        if (testCharacter.isLetter() || (testCharacter == QChar('\"')) || (testCharacter == QChar('\'')))
+        {
+            unitsBufferString += testCharacter;
+
+            previousCharacterWasLetter = true;
+        }
+        else /* if ( ! testCharacter.isLetter()) */
+        {
+            if (previousCharacterWasLetter)
+            {
+                bool conversionSuccessful = false;
+
+                if (!valueBufferString.isEmpty())
+                {
+                    bool ok;
+
+                    double value = valueBufferString.toDouble(&ok);
+
+                    if (ok)
+                    {
+                        RS2::Unit valueUnits = RS_Units::stringToUnit(unitsBufferString);
+
+                        if (valueUnits != RS2::None)
+                        {
+                            if (previousExpressionWasNumber) outputExpression += "+";
```

Comment:
Added the ability to solve, say, `1'2"` in their combined form; it translates to `304.8 + 50.8` in millimetres (mm).